src/main/resources/assets/computercraft/lua/rom/programs/monitor.lua

```diff
@@ -39,7 +39,40 @@ local function resume( ... )
     return param
 end
 
-local ok, param = pcall( function()
+local timers = { n = 0 }
+
+-- add a timer to the timers list with id 'id', xpos 'x', ypos 'y'
+local function addTimer( id, x, y )
+    timers.n = timers.n + 1
```

Comment:
Instead of making this a queue, why not just a map of ids -> positions. This way insertion and removal will be O(1) rather than O(n). Should probably mean the functions can be inlined too.